src/sqlfluff/core/templaters/base.py

```diff
@@ -104,6 +105,29 @@ def __init__(
         self._source_newlines = list(iter_indices_of_newlines(self.source_str))
         self._templated_newlines = list(iter_indices_of_newlines(self.templated_str))
 
+        if check_consistency:
```

Comment:
Can we have a little more explanation on `check_consistency`. Why would you not want it on all the time?

Reply:
Added. Pushing that update now,.

Reply:
Can you open an issue so we don’t forget this? Seems like we don’t really want this option to hang round long term.

Reply:
Done. #2795 2855